homeassistant/components/fronius/sensor.py

```diff
@@ -636,88 +639,92 @@ def _handle_coordinator_update(self) -> None:
 class InverterSensor(_FroniusSensorEntity):
     """Defines a Fronius inverter device sensor entity."""
 
-    coordinator: FroniusInverterUpdateCoordinator
     entity_descriptions = INVERTER_ENTITY_DESCRIPTIONS
 
-    def __init__(self, *args: Any, **kwargs: Any) -> None:
+    def __init__(
+        self,
+        coordinator: FroniusInverterUpdateCoordinator,
+        key: str,
+        solar_net_id: str,
+    ) -> None:
         """Set up an individual Fronius inverter sensor."""
-        super().__init__(*args, **kwargs)
-        self._attr_name = (
-            f"Fronius Inverter {self.solar_net_id} - {self.entity_description.name}"
-        )
-        self._attr_unique_id = (
-            f"{self.coordinator.inverter_info.unique_id}-{self.entity_description.key}"
-        )
+        self._entity_id_prefix = f"inverter_{solar_net_id}"
+        super().__init__(coordinator, key, solar_net_id)
+
+        self._attr_unique_id = f"{coordinator.inverter_info.unique_id}-{key}"
```

Comment:
It would be nice to be able to drop the different sensor entity classes and just have different entity descriptions. We can use a lambda or similar function to calculate the unique_id in the description. Something for the future.

Example: https://github.com/home-assistant/core/blob/9d674af5660a325ed5834c8171bfe8b2144d11b6/homeassistant/components/airly/sensor.py#L105

Reply:
I'll look into it. 
When config_flow is added there are also some DeviceInfo keys to extract, maybe this can be done in the coordinators then.